File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/pigeon_implement/MessageIconDelegateImpl.kt
package com.idosmart.pigeon_implement

import com.idosmart.pigeongen.api_message_icon.MessageIconDelegate

internal class MessageIconDelegateImpl: MessageIconDelegate {

    companion object {
        @Volatile
        private var instance: MessageIconDelegateImpl? = null
        fun instance(): MessageIconDelegateImpl {
            return instance ?: synchronized(this) {
                instance ?: MessageIconDelegateImpl().also { instance = it }
            }
        }
    }

    var updating : Boolean = false
    var dirPath: String = ""

    override fun listenMessageIconState(updating: Boolean) {
       this.updating = updating
    }

    override fun listenIconDirPath(path: String) {
       this.dirPath = path
    }

}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/pigeon_implement/MessageIconDelegateImpl.swift
//
//  MessageIconDelegateImpl.swift
//  protocol_channel
//
//  Created by hc on 2023/11/20.
//

import Foundation

class MessageIconDelegateImpl: MessageIconDelegate {
    
    static let shared = MessageIconDelegateImpl()
    private init() {}
    
    private(set) var updating : Bool = false
    private(set) var dirPath: String = ""
    
    func listenMessageIconState(updating: Bool) throws {
        self.updating = updating
    }
    
    func listenIconDirPath(path: String) throws {
        self.dirPath = path
    }
    
    
}
